`@types/react-dom` incorrectly declared as a `devDependency`

- `@testing-library/react` version: v12.1.2
- Testing Framework and version: N/A
- DOM Environment: N/A

### What you did:

Tried to use `@testing-library/react` in a TS web app built with Bazel's [rules_nodejs](https://github.com/bazelbuild/rules_nodejs), which enforce strict visibility between NPM packages as determined by their declared dependencies.

### What happened:

Got the following tsc error when compiling a test that uses `@testing-library/react` with `rules_nodejs`:
```
node_modules/@testing-library/react/types/index.d.ts(9,24): error
TS2307: Cannot find module 'react-dom' or its corresponding type
declarations.
node_modules/@testing-library/react/types/index.d.ts(10,31): error
TS2307: Cannot find module 'react-dom/test-utils' or its corresponding
type declarations.
```

### Problem description:

`@testing-library/react` declares `@types/react-dom` as a [`devDependency`](https://github.com/testing-library/react-testing-library/blob/8f17a2bc4fa9c2c17a623b38ded9ab18b086e457/package.json#L52), but uses it in [its external interface](https://github.com/testing-library/react-testing-library/blob/8f17a2bc4fa9c2c17a623b38ded9ab18b086e457/types/index.d.ts#L9) when used from TypeScript.

### Suggested solution:

Mark `@types/react-dom` as a regular dependency of `@testing-library/react` so that packages depending on `@testing-library/react` also pick up this dependency.

--- a/package.json
+++ b/package.json
@@ -45,11 +45,11 @@
   "license": "MIT",
   "dependencies": {
     "@babel/runtime": "^7.12.5",
-    "@testing-library/dom": "^8.0.0"
+    "@testing-library/dom": "^8.0.0",
+    "@types/react-dom": "^17.0.0"
   },
   "devDependencies": {
     "@testing-library/jest-dom": "^5.11.6",
-    "@types/react-dom": "^17.0.0",
     "dotenv-cli": "^4.0.0",
     "kcd-scripts": "^11.1.0",
     "npm-run-all": "^4.1.5",

--- a/package.json
+++ b/package.json
@@ -46,7 +46,7 @@
   "dependencies": {
     "@babel/runtime": "^7.12.5",
     "@testing-library/dom": "^8.0.0",
-    "@types/react-dom": "^17.0.0"
+    "@types/react-dom": "*"
   },
   "devDependencies": {
     "@testing-library/jest-dom": "^5.11.6",

